Places/View Controllers/ItineraryDetailViewController.m

```diff
@@ -34,6 +35,49 @@ - (void)viewDidLoad {
     self.placesToGoTableView.dataSource = self;
 }
 
+- (void)presentActivityController:(UIActivityViewController *)controller {
+
+    // for iPad: make the presentation a Popover
+    controller.modalPresentationStyle = UIModalPresentationPopover;
+    [self presentViewController:controller animated:YES completion:nil];
+
+    UIPopoverPresentationController *popController = [controller popoverPresentationController];
+    popController.permittedArrowDirections = UIPopoverArrowDirectionAny;
+    popController.barButtonItem = self.navigationItem.leftBarButtonItem;
+
+    // access the completion handler
+    controller.completionWithItemsHandler = ^(NSString *activityType,
+                                              BOOL completed,
+                                              NSArray *returnedItems,
+                                              NSError *error){
+        // react to the completion
+        if (completed) {
+            // user shared an item
+            NSLog(@"We used activity type%@", activityType);
+        } else {
+            // user cancelled
+            NSLog(@"We didn't want to share anything after all.");
+        }
+
+        if (error) {
+            NSLog(@"An Error occured: %@, %@", error.localizedDescription, error.localizedFailureReason);
+        }
```

Comment:
Maybe refactor this if block to account for the 3 possibilities:
1. completed
2. canceled
3. failed

Something like this:
```
if (completed) { // shared
} else if (error == nil) { // canceled
} else { // error
```

With your current implementation, in case of an error, your log will say:
```
We didn't want to share anything after all.
An Error occured: <error>
```

Reply:
Additionally, it might be a good idea to add a completion block to the this method that allows its callers to get notified when the user has finished sharing an itinerary (including any errors that have happened). This allows the caller to handle the error if needed, e.g. by showing an alert to the user.